### ISSUE TITLE
Execution failed for task ':app:checkReleaseDuplicateClasses' for android

### ISSUE THREAD

**Issue by vr7bd** [Rapsssito/react-native-tcp-socket#199]
## Description
After upgrading to 6.2.0, I'm unable to build my project with this error
```
Execution failed for task ':app:checkReleaseDuplicateClasses'.

> A failure occurred while executing com.android.build.gradle.internal.tasks.CheckDuplicatesRunnable

   > Duplicate class org.bouncycastle.LICENSE found in modules bcprov-jdk15on-1.70.jar -> jetified-bcprov-jdk15on-1.70 (org.bouncycastle:bcprov-jdk15on:1.70) and bcprov-jdk15to18-1.70.jar -> jetified-bcprov-jdk15to18-1.70 (org.bouncycastle:bcprov-jdk15to18:1.70)
```
I traced it and found out that this occurs if `expo-updates` is installed and `react-native-tcp-socket` is on version `6.2.0`. It's because of the commit 054c789 where these dependencies are added
```
    // Bouncy Castle dependencies
    implementation 'org.bouncycastle:bcprov-jdk15on:1.70'
    implementation 'org.bouncycastle:bcpkix-jdk15on:1.70'
```

## Steps to reproduce
1. Create a new expo project and install expo updates
2. Install the latest version of react-native-tcp-socket
3. `npx expo prebuild && npx expo run:android`

**Comment by siarheipashkevich** [comment]
The same issue!

@regulad how to do your solution for Expo EAS builds?

**Comment by siarheipashkevich** [comment]
@vr7bd did you solve this issue?

**Comment by vr7bd** [comment]
> @vr7bd did you solve this issue?

Not exactly. This [works](https://github.com/Rapsssito/react-native-tcp-socket/issues/199#issuecomment-2308919311) but as you pointed out, for EAS builds, this is not a good solution. So I just downgraded to `6.0.1` (if I remember correctly), which is the version prior to the commit of the dependency

**Comment by ShaneZhengNZ** [comment]
This can be solved by a custom Expo config plugin. I have done this for my project. I can open source it.

**Comment by siarheipashkevich** [comment]
@ShaneZhengNZ please share it if it is possible

**Comment by ShaneZhengNZ** [comment]
@siarheipashkevich sure thing. I should have this done in the next a few days. I will pin you when the package is available to use.

**Comment by ShaneZhengNZ** [comment]
[with-react-native-tcp-socket](https://www.npmjs.com/package/with-react-native-tcp-socket)

@vr7bd @siarheipashkevich

**Comment by siarheipashkevich** [comment]
@ShaneZhengNZ thanks! How do you think is it possible to "patch" this package for avoiding this issue using https://www.npmjs.com/package/patch-package?

**Comment by ShaneZhengNZ** [comment]
I have used yarn patch (similar thing) for patching javascript code or native code before, but never done the build.gradle.
Worth a shot I guess?

**Comment by jakobbuis** [comment]
I just ran across this while making a production build (and installing expo-updates). Very much still a thing, glad I found this topic. 

I reverted to 6.0.1. as a solution.

**Comment by jakobbuis** [comment]
Reverting didn't work for me, so I updated the dependency. Made a PR to merge #205. 

In the meantime, you can use `patch-package` to apply the fix as long as the PR isn't merged.
